lib/models/BucketPolicy.ts

```diff
@@ -48,64 +48,77 @@ const objectActions = [
     's3:PutObjectTagging',
 ];
 
+export type BucketPolicyMetadata = {
+    Id: string;
+    Version: string;
+    Statement: {
+        Sid: string;
+        Effect: string;
+        Principal: string | { AWS: string[] };
```

Comment:
```suggestion
        Principal: string | { AWS: string | string[] };
```

This can be a string as well like `Principal: { AWS: '*' }`